e2e/tests/upstreams.crud-all-fields.spec.ts

```diff
@@ -0,0 +1,542 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { tls } from '@e2e/fixtures/tls';
+import { upstreamsPom } from '@e2e/pom/upstreams';
+import { randomId } from '@e2e/utils/common';
+import { e2eReq } from '@e2e/utils/req';
+import { test } from '@e2e/utils/test';
+import { uiHasToastMsg } from '@e2e/utils/ui';
+import { expect, type Locator } from '@playwright/test';
+
+import { deleteAllUpstreams } from '@/apis/upstreams';
+
+test.beforeAll(async () => {
+  await deleteAllUpstreams(e2eReq);
+});
+
+test('should CRUD upstream with all fields', async ({ page }) => {
+  test.setTimeout(30000);
+
+  const fillHTTPStatuses = async (input: Locator, ...statuses: string[]) => {
+    for (const status of statuses) {
+      await input.fill(status);
+      await input.press('Enter');
+    }
+  };
+
+  const upstreamNameWithAllFields = randomId('test-upstream-full');
+  const description =
+    'This is a test description for the upstream with all fields';
+
+  // Navigate to the upstream list page
+  await upstreamsPom.toIndex(page);
+  await upstreamsPom.isIndexPage(page);
+
+  // Click the add upstream button
+  await upstreamsPom.getAddUpstreamBtn(page).click();
+  await upstreamsPom.isAddPage(page);
+
+  await test.step('fill in required fields', async () => {
+    // Fill in the required fields
+    // 1. Name (required)
+    await page
+      .getByLabel('Name', { exact: true })
+      .fill(upstreamNameWithAllFields);
+
+    // 2. Description (optional but simple)
+    await page.getByLabel('Description').fill(description);
+
+    // 3. Add multiple nodes (required)
+    const addNodeBtn = page.getByRole('button', { name: 'Add a Node' });
+    const nodesSection = page.getByRole('group', { name: 'Nodes' });
+
+    // Wait for 'No Data' text to be visible
+    const noData = nodesSection.getByText('No Data');
+    await expect(noData).toBeVisible();
+
+    // Add the first node, using force option
+    await addNodeBtn.click();
+    await expect(noData).toBeHidden();
+
+    // Wait for table rows to appear
+    const rows = nodesSection.locator('tr.ant-table-row');
+    await expect(rows.first()).toBeVisible();
+
+    // Fill in the Host for the first node - click first then fill
+    const hostInput = rows.first().locator('input').first();
+    await hostInput.click();
+    await hostInput.fill('node1.example.com');
+    await expect(hostInput).toHaveValue('node1.example.com');
```

Comment:
Where to test the validation rules of the form?

Reply:
Currently, there are no validation rules. Aside from the simplest ones like numbers and strings, data is directly submitted to APISIX for processing.